protobuf/signalling.proto

```diff
@@ -160,10 +162,36 @@ message ClientToStation {
     // YIELD=>ACQUIRE switchover is happening.
     optional uint64 upload_sync = 4;
 
-    // 
+
+    // List of decoys that client have unsuccessfully tried in current session.
+    // Could be sent in chunks
     repeated string failed_decoys = 10;
 
+    optional SessionStats stats = 11;
+
+    // Station is only required to check this variable during session initialization.
+    // If set, station must facilitate connection to said target by itself, i.e. write into squid
+    // socket an HTTP/SOCKS/any other connection request.
+    // covert_address must have exactly one ':' colon, that separates host (literal IP address or
+    // resolvable hostname) and port
+    // TODO: make it required for initialization, and stop connecting any client straight to squid?
+    optional string covert_address = 20;
+
     // Random-sized junk to defeat packet size fingerprinting.
     optional bytes padding = 100;
 }
 
+message SessionStats {
+    optional uint32 failed_decoys = 20; // how many decoys were tried before success
+
+    // Timings below are in nanoseconds
+
+    // Applicable to whole session:
+    optional int64 total_time_to_connect = 31; // includes failed attempts
```

Comment:
uint32? use microseconds? Max time: 1 hour 6 minutes....

Reply:
That would work too. If we are reducing precision though, milliseconds might be a safer choice? We may eventually want to get some kind of sessions-wide metric "Hey, I've been trying to connect for the whole day"